### ISSUE TITLE
ParallelScope None does not work consistently on parameterized tests

### ISSUE THREAD

**Issue by Riverlution** [nunit/nunit#3992]
I look after a test suite that is designed to run all tests in parallel, except if they are marked otherwise. All tests inherit from a base class with the ParallelizableAttribute applied, taking the ParallelScope.All parameter. On tests that we do not want to run in parallel, we mark them with [Parallelizable(ParallelScope.None)], and this works fine for non-parameterized tests. It seems to be a different story for Parameterized tests that are marked with [Parallelizable(ParallelScope.None)]

I have managed to replicate the issue with the code below:

```
[Parallelizable(ParallelScope.All)]
public class TestsIncludingParameterizedTestsThatCantRunInParallel
{
	private static readonly object LockObj = new object();
	private static readonly TimeSpan Timeout = TimeSpan.FromSeconds(1);

	/// <summary>
	/// Try to perform a task in parallel
	/// </summary>
	/// <returns>True if the caller could acquire the lock and perform the task, false if another caller already had a lock</returns>
	public bool PerformParallelUnsafeTask()
	{
		bool lockTaken = false;
		try {
			Monitor.TryEnter(LockObj, 1, ref lockTaken);
			if (lockTaken)
			{
				Thread.Sleep(Timeout);
				return true;
			}
			else {
				// The lock was not acquired.
				return false;
			}
		}
		finally {
			// Ensure that the lock is released.
			if (lockTaken) {
				Monitor.Exit(LockObj);
			}
		}
	}
	
	public void AssertTaskWasNotRanInParallel()
	{
		Assert.That(PerformParallelUnsafeTask(), $"Task was performed in parallel");
	}

	[Test]
	public void TestA()
	{
		AssertTaskWasNotRanInParallel();
	}

	[TestCase(1)]
	[TestCase(2)]
	[Parallelizable(ParallelScope.None)]
	public void TestB(int val)
	{
		AssertTaskWasNotRanInParallel();
	}
}
```

When I run the above with parallelisation turned on, some tests always fail
I usually see that TestA passes by itself, and TestB(1)/TestB(2) both fail:
![image](https://user-images.githubusercontent.com/16063171/141975310-b8bd9620-77dc-441c-9e43-f5ff57bef90c.png)

Unless I have misunderstood how these parallelizable attributes and their inheritance should work, TestB should not run in parallel with any other tests. It should not matter that TestA is not marked to not run in parallel, surely it should only be ran alongside other tests that we have marked to run in parallel?

Oddly, I found that everything works fine if I change the name of the tests. If I change TestA to TestD, it looks like everything passes (no tests are actually ran in parallel). 
![image](https://user-images.githubusercontent.com/16063171/141976345-fc8dc947-2bab-43b4-b104-ed80e71c93d0.png)

Replicated with NUnit 3.13.2 and NUnit3TestAdapter 4.1.0

**Comment by mark-ashton** [comment]
I am seeing the same problem, but with the `[NonParallelizable]` attribute which I assume works in the same way as `[Parallelizable(ParallelScope.None)]`
If I change the test to be non-parameterised then it will respect the NonParallelizable, and run it after all other parallel tests are complete. When parameterised with `[TestCase]` though the NonParallelizable is ignored and it is run at the same time as other tests.

**Comment by stevenaw** [comment]
A few parallelization fixes to 3.13 bugs have landed in main recently. I'm going to label this confirm to see if it's still happening

**Comment by stevenaw** [comment]
Oh, actually this seems to be a duplicate of #3874
